src/averageAges.js

```diff
@@ -37,7 +38,20 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const getAverageWomen = (!withChildren)
+    ? people.filter(gender => gender.sex === 'f')
+    : people.filter(mother => mother.sex === 'f'
+    && (people.some(person => mother.name === person.mother)));
+
+  return getAverage(getAverageWomen, 'died', 'born');
+}
+
+function getAverage(people, value1, value2) {
+  const averageAge = people
+    .reduce((sum, { [value1]: diedYear, [value2]: bornYear }) =>
+      sum + (diedYear - bornYear), 0);
```

Comment:
[checklist point 2](https://github.com/mate-academy/js_average-ages/blob/master/checklist.md)

```suggestion
  const averageAge = people
    .reduce((sum, { [value1]: diedYear, [value2]: bornYear }) => (
       sum + (diedYear - bornYear)
     ), 0);
```